experiments/ClimaEarth/user_io/io_helpers.jl

```diff
@@ -14,24 +16,24 @@ Create output directories for the experiment. If `comms_ctx` is provided, only t
 # Returns
 - A tuple with the paths to the output, regrid, and artifacts directories.
 """
-function setup_output_dirs(; output_dir = nothing, regrid_dir = nothing, artifacts_dir = nothing, comms_ctx)
+function setup_output_dirs(; output_dir = nothing, artifacts_dir = nothing, comms_ctx)
     if output_dir === nothing
         output_dir = "."
     end
-    if regrid_dir === nothing
-        regrid_dir = joinpath(output_dir, "regrid_tmp/")
-    end
     if artifacts_dir === nothing
         artifacts_dir = output_dir * "_artifacts"
     end
 
     @info(output_dir)
+    regrid_dir = nothing
     if ClimaComms.iamroot(comms_ctx)
         mkpath(output_dir)
-        mkpath(regrid_dir)
         mkpath(artifacts_dir)
+        regrid_dir = mktempdir(output_dir, prefix = "regrid_tmp_")
     end
 
+    ClimaComms.barrier(comms_ctx)
+    regrid_dir = ClimaComms.bcast(comms_ctx, regrid_dir)
```

Comment:
`MPI.Bcast` implies barriers (as opposed to `MPI.Ibcast` which is the non-blocking version), so you don't need barriers here